### PR TITLE
MEN-4503: track the campaign parameter in the sign up page

### PR DESCRIPTION
**Pull request by tranchitella**
Changelog: none

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

**Comment by tranchitella** [comment]
Depends on: https://github.com/mendersoftware/tenantadm/pull/276

**Comment by tranchitella** [comment]
@mzedel I have no idea why unit tests are failing in components I haven't touched. Ideas?

**Comment by mzedel** [comment]
yep, they weren't touched for a while, but fail suddenly consistently on Gitlab (while passing locally)... https://github.com/mendersoftware/gui/pull/1507 tries to fix this, but the underlying reason is still a mystery to me...

**Comment by tranchitella** [comment]
@mzedel shall we ignore that specific failure? the PR is ready-for review, my change here is minimal.

File: src/js/components/user-management/signup.js [hunk 1]
```diff
@@ -27,6 +27,7 @@ export const Signup = ({ createOrganizationTrial, currentUserId, loginUser, setF
   const [organization, setOrganization] = useState('');
   const [tos, setTos] = useState(false);
   const [redirectOnLogin, setRedirectOnLogin] = useState(false);
+  const campaign = match?.params?.campaign;
```
Comment (mzedel) [inline src/js/components/user-management/signup.js:30]: `match` and `params` should always be defined, due to the router wrapping the `Signup` component - but these conditions won't hurt anyone 🙂

**Review by mzedel — APPROVED** [review]

File: src/js/components/user-management/signup.js (hunk 1 above)
Comment (tranchitella) [reply src/js/components/user-management/signup.js:30]: @mzedel they are not in unit tests, though... can I fix it somehow?

File: src/js/components/user-management/signup.test.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ describe('Signup Component', () => {
     const { baseElement } = render(
       <MemoryRouter>
         <Provider store={store}>
-          <Signup location={{ state: { from: '' } }} />
+          <Signup location={{ state: { from: '' }, match: { params: { campaign: '' } } }} />
```
Comment (mzedel) [inline src/js/components/user-management/signup.test.js:22]: @tranchitella the location is a misleading copy&paste remnant from the login component test
```suggestion
          <Signup match={{ params: { campaign: '' } }} />
```
should work & allow the access to the match without conditions

Comment (tranchitella) [reply src/js/components/user-management/signup.test.js]: Thanks!

**Pull request merged**